doc/source/reference/c-api.array.rst

```diff
@@ -3466,6 +3466,10 @@ Other constants
 
     The maximum number of dimensions allowed in arrays.
 
+.. c:var:: NPY_MAXARGS
+
+    The maximum number of arguments that can be used in arrays.
```

Comment:
I see how this one is confusing. It is the maximum number of arguments to many numpy functions. I.e. the maximum number of arrays that can be processed at once. That is most notably the iterator (which should limit things such as `np.choose`).